tests/validation/tests/v3_api/test_cluster_templates.py

```diff
@@ -361,6 +361,159 @@ def test_cluster_template_add_readonly_member():
+        get_client_for_token(user_token["user_standard"]["token"])
+
+    rke_config = getRKEConfig(k8sversionlist[0])
+
+    cluster_name = random_test_name("test-auto-import")
+    cluster = create_node_cluster(standard_user_client, cluster_name,
+                                  rancherKubernetesEngineConfig=rke_config)
+
+    # Export a Template
+    cluster.saveAsTemplate(clusterTemplateName="testnewrketemplate",
+                           clusterTemplateRevisionName="v1")
+    cluster = standard_user_client.reload(cluster)
+    templateid = cluster.clusterTemplateId
+    revisionid = cluster.clusterTemplateRevisionId
+
+    # Create a new cluster using the template revision just exported
+    newcluster = create_node_cluster(
+        standard_user_client, name=random_test_name("test-auto"),
+        clusterTemplateRevisionId=revisionid)
+    newcluster = standard_user_client.reload(newcluster)
+    assert newcluster.appliedSpec.clusterTemplateId == templateid
+    assert newcluster.appliedSpec.clusterTemplateRevisionId == revisionid
+
+    cluster_cleanup(standard_user_client, cluster)
+    cluster_cleanup(standard_user_client, newcluster)
+
+
+def test_cluster_template_enforcement_on_admin():
+
+    # As an admin turn ON enforcement and ensure that admin can create clusters
+    # using rke config and also using rke template
+
+    enforcement_settings_url = CATTLE_TEST_URL + \
+                      "/v3/settings/cluster-template-enforcement"
+    data_test = {
+        "name": "cluster-template-enforcement",
+        "value": "true"
+    }
+    headers = {"Content-Type": "application/json",
+               "Accept": "application/json",
+               "Authorization": "Bearer " + ADMIN_TOKEN}
+    response = requests.put(enforcement_settings_url, json=data_test,
+                            verify=False, headers=headers)
+    print(response.content)
+    k8sversionlist = get_k8s_versionlist()
+    cluster_config1 = get_cluster_config(k8sversionlist[0])
+    rke_config = getRKEConfig(k8sversionlist[0])
+
+    # Verify creating cluster using rkeconfig succeeds
+
+    client = get_admin_client()
+    cluster_name = random_test_name("test-auto-rkeconfig")
+    rkecluster = create_node_cluster(client, cluster_name,
+                                     rancherKubernetesEngineConfig=rke_config)
+
+    # Verify creating cluster using rke template succeeds
+    cluster_template = client.create_cluster_template(
+        name=random_test_name("template"), description="test-template")
+
+    cluster_template_revision1 = client.create_cluster_template_revision(
+        name=random_test_name("revision1"),
+        clusterConfig=cluster_config1,
+        clusterTemplateId=cluster_template.id)
+
+    cluster_name = random_test_name("test-auto")
+    cluster = create_node_cluster(
+        client, name=cluster_name,
+        clusterTemplateRevisionId=cluster_template_revision1.id)
+    check_cluster_version(cluster, k8sversionlist[0])
+
```

Comment:
This block of code should be executed even when failure are encountered during the execution of this test.

Reply:
Added try and finally block.